scripts/linters/python_linter.py

```diff
@@ -336,19 +302,18 @@ def lint_py_files(self):
             current_files_to_lint = files_to_lint[
                 current_batch_start_index: current_batch_end_index]
 
-            # This line invokes Pylint and redirect its output
-            # to the StringMessageStream.
-            pylint_report = StringMessageStream()
+            pylint_report = python_utils.string_io()
             pylinter = lint.Run(
                 current_files_to_lint + [config_pylint],
                 reporter=text.TextReporter(pylint_report),
                 exit=False).linter
 
             if pylinter.msg_status != 0:
-                for message in pylint_report.read():
-                    full_error_messages.append(message)
+                lint_message = pylint_report.getvalue()
+                full_error_messages.append(lint_message)
+
                 pylint_error_messages = (
-                    self.get_trimmed_error_output(pylint_report.read()))
```

Comment:
It was expected to use `string_io` instead of `StringMessageStream` but while implementing this Anshul was facing an issue as this read call wasn't returning the error log so he ended up creating a separate class for capturing the lint logs. 

I found that it (read call not returning logs) was happening because the first read call on line 438 changes the cursor to the end of the message and it was expected to seek(0) before calling read again.